### ISSUE TITLE
Connect to local "remote" existing kernel (for Docker)

### ISSUE THREAD

**Issue by tiangolo** [nteract/hydrogen#287]
It would be great to have a way to make Hydrogen connect to an existing local Kernel using an existing connection file, like in:

`jupyter console --existing /home/user/test-docker-python/tmp/kernel/kernel-000.json`

This would allow developing in a Docker container with all its dependencies but having autocompletion, code run, etc.

With that, it would be possible to start a Docker container with all of its dependencies installed, mounting a volume to a local directory and starting the IPython kernel (from within the container) with the file in that directory. And then running the code from Atom using that Kernel.

Of course, this would leave to the user starting the Kernel the creation of the connection file, etc. And also every security concerns would be up to the user. But having a local Docker environment there wouldn't be any security concerns.

**Comment by slavaGanzin** [comment]
@tiangolo try 
``` jupyter kernelspec list ``` or ``` iptyhon kernelspec list```
You'll see something like this:
```
Available kernels:
  coffeescript       /home/vganzin/.local/share/jupyter/kernels/coffeescript
  javascript         /home/vganzin/.local/share/jupyter/kernels/javascript
  python2-optirun    /home/vganzin/.local/share/jupyter/kernels/python2-optirun
  python2            /usr/local/share/jupyter/kernels/python2
  python3            /usr/local/share/jupyter/kernels/python3

```

just make a symlink to one of this folders like:

```
ln -s /home/user/test-docker-python/tmp/kernel/kernel-000.json /home/user/.local/share/jupyter/kernels/kernel.json

```

And trigger 'hydrogen update kernels' it should work

**Comment by tiangolo** [comment]
Thanks @slavaGanzin for the tip.

Here's what happens.

When I run:

```
$ jupyter kernelspec list
```

I get:

```
Available kernels:
  python2    /home/user/miniconda2/lib/python2.7/site-packages/ipykernel/resources
```

(I'm using miniconda / Anaconda Python).

But that folder has no contents (just two images):

```
$ ls /home/user/miniconda2/lib/python2.7/site-packages/ipykernel/resources 
logo-32x32.png  logo-64x64.png
```

Nevertheless, it seems that [recently the location has moved](http://stackoverflow.com/a/36433438), and to get it is with:

```
$ jupyter --runtime-dir
/run/user/1000/jupyter
```

And inside it is the kernel connection file:

```
ls /run/user/1000/jupyter/       
kernel-16604.json
```

And the contents are the connection parameters as expected (not a kernel specification or something like that):

```
cat /run/user/1000/jupyter/kernel-16604.json                                                 
{
  "stdin_port": 39753, 
  "ip": "127.0.0.1", 
  "control_port": 44910, 
  "hb_port": 33512, 
  "signature_scheme": "hmac-sha256", 
  "key": "64562052-82fb-4022-b93a-d7131f75df7c", 
  "kernel_name": "", 
  "shell_port": 46955, 
  "transport": "tcp", 
  "iopub_port": 59018
}
```

But then I run "hydrogen update kernels" and "hydrogen show kernel commands" and in the submenu I only get "Switch to Python 2" (apart "Interrupt" and "Restart").

The thing is that no matter what I do, Hydrogen never uses any of those connection files. It always creates a new connection file to a new kernel inside the installation directory inside Atom, for example in:

```
/home/user/.atom/packages/Hydrogen/kernel-configs/kernel-fecc4fea-7b53-4599-9fb2-45fd96ad4da9.json
```

which is a new connection file:

```
cat /home/user/.atom/packages/Hydrogen/kernel-configs/kernel-fecc4fea-7b53-4599-9fb2-45fd96ad4da9.json
{
  "stdin_port": 49808, 
  "ip": "127.0.0.1", 
  "control_port": 45350, 
  "hb_port": 58863, 
  "signature_scheme": "hmac-sha256", 
  "key": "b6989c89-6896-4572-a677-b0c7fac8c37c", 
  "kernel_name": "", 
  "shell_port": 50343, 
  "transport": "tcp", 
  "iopub_port": 59836
}
```

(It's shown in the Atom Developer Tools console every time).

---

I think that a simple solution could be adding a simple behavior that checks in the current project directory for a file like `./hydrogen/kernel/custom-kernel.json` and if it exists then uses it instead of creating a completely new kernel. I think it wouldn't be too complex to implement, you have already made all the hard stuff in Hydrogen.

I don't know if that is something that you see fit in the project, but maybe I could try adding it (although I'm not an expert in CoffeScript).

For me (and I think for several people, at least in my team), that would make Atom with Hydrogen the default IDE for Python in Docker development (at least). But I don't know what's your opinion about it, is that a feature that you would like (or accept) to have?

**Comment by slavaGanzin** [comment]
> I think that a simple solution could be adding a simple behavior that checks in the current project directory for a file like ./hydrogen/kernel/custom-kernel.json and if it exists then uses it instead of creating a completely new kernel. I think it wouldn't be too complex to implement, you have already made all the hard stuff in Hydrogen.

looks like great idea

**Comment by lgeiger** [comment]
I have never used a remote kernel but this sounds like a great addition to hydrogen!

> but maybe I could try adding it (although I'm not an expert in CoffeScript).

Awesome! This would be great! 🎉

**Comment by n-riesco** [comment]
Closed via #295
